image-cropper/src/app/cropper/selection/rect_selection.rs

```diff
@@ -0,0 +1,333 @@
+                        color: Color::grayscale_f01(0.5),
+                        width: 1.0,
+                        border_position: BorderPosition::Inside,
+                    }),
+                    ..Default::default()
+                },
+            })
+            .attach_event(|builder| {
+                let id = self.id.clone();
+                builder.on_mouse_down(move |event| {
+                    if event.pressing_buttons.contains(&namui::MouseButton::Right) {
+                        namui::event::send(SelectionEvent::SelectionRightClicked {
+                            target_id: id.clone(),
+                        })
+                    }
+                })
+            }),
+            render_handles(&scaled_xywh, self.id.clone()),
+        ])
+    }
+
+    fn get_polygon(&self) -> Vec<namui::Xy<f32>> {
+        vec![
+            Xy {
+                x: self.xywh.x,
+                y: self.xywh.y,
+            },
+            Xy {
+                x: self.xywh.x + self.xywh.width,
+                y: self.xywh.y,
+            },
+            Xy {
+                x: self.xywh.x + self.xywh.width,
+                y: self.xywh.y + self.xywh.height,
+            },
+            Xy {
+                x: self.xywh.x,
+                y: self.xywh.y + self.xywh.height,
+            },
+        ]
+    }
+
+    fn get_id(&self) -> &String {
+        &self.id
+    }
+}
+
+#[derive(Clone)]
+struct RectSelectionResizeHandle {
+    pub handle_direction: RectSelectionResizeDirection,
+    pub polygon_xy: Vec<Xy<f32>>,
+    pub cursor: MouseCursor,
+}
+fn render_handles(xywh: &XywhRect<f32>, selection_id: String) -> RenderingTree {
+    render(get_handles(xywh).iter().map(|handle| {
+        let selection_id = selection_id.clone();
+        let direction = handle.handle_direction.clone();
+        let path = PathBuilder::new().add_poly(&handle.polygon_xy, true);
+
+        let stroke_paint = PaintBuilder::new()
+            .set_style(PaintStyle::Stroke)
+            .set_stroke_width(1.0)
+            .set_color(Color::WHITE);
+
+        let fill_paint = PaintBuilder::new()
+            .set_style(PaintStyle::Fill)
+            .set_color(Color::BLACK);
+
+        render([
+            namui::path(path.clone(), fill_paint),
+            namui::path(path, stroke_paint),
+        ])
+        .with_mouse_cursor(handle.cursor)
+        .attach_event(move |builder| {
+            let selection_id = selection_id.clone();
+            let direction = direction.clone();
+            builder.on_mouse_down(move |_| {
+                let selection_id = selection_id.clone();
+                let direction = direction.clone();
+                namui::event::send(SelectionEvent::RectSelectionResizeHandleClicked {
+                    selection_id,
+                    direction,
+                })
+            })
+        })
+    }))
+}
+
+fn get_handles(xywh: &XywhRect<f32>) -> Vec<RectSelectionResizeHandle> {
```

Comment:
From [this code](https://github.com/NamseEnt/namseent/blob/master/luda-editor/luda-editor-client/src/app/editor/clip_editor/camera_clip_editor/wysiwyg_editor/character_wysiwyg_editor/cropper.rs#L98)

Thank you Namse!

Reply:
Namse Can Do Anything